controllers/openstackcluster_controller.go

```diff
@@ -290,7 +290,18 @@ func deleteBastion(scope *scope.WithLogger, cluster *clusterv1.Cluster, openStac
 		}
 	}
 
-	if instanceStatus != nil {
+	// If no instance was created we currently need to check for orphaned
+	// volumes. This requires resolving the instance spec.
+	// TODO: write volumes to status resources on creation so this is no longer required.
```

Comment:
The bastion flow here is not as robust as the machine flow. However, this was the case before and I haven't tried as hard to fix it. The reason is that the bastion delete flow is extremely janky and over-complicated, and what I really want to do is re-design it.

As long as I didn't regress it here I'm ok with not fixing all the edge cases.

Reply:
Incidentally, notice that previously because we only ran any of this when `instanceStatus != nil` we would always leak volume in this case. With this change we no longer leak volumes. Updating volumes to use the same pattern as ports will make this simpler, but that's for another time.

Reply:
I'm just commenting on a comment here, I don't think we'll be able to get rid of the `DeleteVolumes()` even after volumes are saved into the ID.

Reply:
Well... kinda. I eventually want to move them into resources, and at that point they'll go into the Adopt() flow.